### DISCUSSION TITLE
Running games in a container

### DISCUSSION THREAD

**Discussion by svin24** [jc141x/jc141-bash#20] [Show and tell]
### Requirements

- podman
- distrobox

### Notes:

- So far only tested with an arch container.
- Only tested out Elden Ring.
- Performance seems fine at least on my laptop with a ryzen APU. I have not done any in depth tests.
- Shaders probably will need recompilation. So expect stutters for the first few seconds
- Saves carry over obviously

### Why?

Because i don't like installing 32-bit libraries on my system. Nor do i want to use raw wine installed.

### Potential benefits of this approach

You could have a generic image in docker everyone can use immediately, possibly integrate it with Chad client. Most incompatibilities should go away.

### Loose instructions

1. Create a distrobox container with archlinux and then enter

``` 
distrobox create -n gamecontainer -i docker.io/archlinux:latest
distrobox enter gamecontainer
```

2. Follow the [setup guide for archlinux](https://github.com/jc141x/jc141-bash/blob/master/setup/arch.md).

Not too faithfully as it seems there is an issue with the oneliner that enables multilib or something
Just enable multilib and the rumpowered repo manually using nano or vim, you can even use a graphical program if you hate those.

You might figure out additional stuff installed too. Depending on the game maybe.

3. run the game

Navigate to the directory and run it.

My Host: Fedora
My container: Archlinux

![image](https://github.com/jc141x/jc141-bash/assets/51864332/3bd1becf-4d9b-4ab5-a2b0-7603b78c47d0)

### NOT TESTED

- NVIDIA, i don't have any NVIDIA cards.
- Steam for steam controller integration, although i don't see why it would not work (you can export applications to the main system with distrobox which is something i already do for VScode)
- Many other games
- Docker instead of podman

**Comment by adminx01** [comment]
> possibly integrate it with Chad client

It's not an ongoing project. Not enough people to work on it. Other projects have more resources for creating friendly user interfaces.

**Comment by hackernoodle** [comment]
what is "raw wine"

**Reply by svin24** [reply]
A wine package that is not containerized or managed by something other than your system package manager.

**Reply by hackernoodle** [reply]
all of our setup guides use the system package manager to install wine.

> `sudo pacman -S --needed wine-staging`
> `sudo apt install winehq-staging`
> `sudo dnf install wine`
> `sudo nix-shell -p wine-staging`
> `sudo zypper install wine-staging`

**Reply by hackernoodle** [reply]
because of drawfs, deb based systems will need a third party packaging tool called [makedeb](https://www.makedeb.org/). it's based on makepkg for arch that enables you to use the AUR.

using an MPR helper (a wrapper around makedeb) like [una](https://github.com/AFK-OS/una) will give automatic updates for the makedeb version, however it has been too buggy for us to honestly recommend it

**Reply by svin24** [reply]
yeah but what if that stuff did not need to touch your system and did not force you to go though more frequent upgrades

Also as for makedeb all i have to say is [this](https://wiki.debian.org/DontBreakDebian)

**Reply by hackernoodle** [reply]
> Also as for makedeb all i have to say is [this](https://wiki.debian.org/DontBreakDebian)

fair point, but debian is already broken. if it wasn't so shitty to get user content available to apt we wouldn't have this problem

**Comment by KupchykVitaly** [comment]
Hi can anyone help with installing mods on total war warhammer 3 repack joncena 141 v2.4.0

In the launcher, I open the settings folder, find the local_mods folder (/home/xxxxx/.local/share/feral-interactive/Total War WARHAMMER III/VFS/Local/local_mods), dump the tww3cheatmod.pack mod, and then check out the workshop Steam and there is nothing there, help to install :(

**Reply by hackernoodle** [reply]
join the [matrix rooms](https://matrix.to/#/#rumpowered:matrix.org), lots of friendly people ready to help

**Reply by KupchykVitaly** [reply]
Thanks for help
